client/app/views/contacts/index.coffee

```diff
@@ -63,65 +73,97 @@ module.exports = class Contacts extends Mn.CompositeView
+
+        if @model?.has 'char'
+            @collection = app.filtered
+            @hasContacts = true
+
+            @filter = (vmodel) =>
+                _.contains @collection.indexes.get(@model.get('char')), vmodel
 
-        else unless @model or app.model.get 'scored'
-            collection = @collection
-            groups = [].map.call 'abcdefghijklmnopqrstuvwxyz#', (char) ->
-                group = new Backbone.Model name: char
-                group._attachedCollection = collection
-                return group
-            @collection = new Backbone.Collection groups
+            @listenTo @collection,
+                'update': ->
+                    @updateTagClasses()
+                    # @toggleEmpty()
+
+            @listenTo @collection.underlying,
+                'change:tags': ->
+                    @updateTagClasses()
+                # 'add': ->
+                #     @toggleEmpty()
+                # 'remove': ->
+                #     @toggleEmpty()
+
+        else unless @indexed
+            @collection = new Backbone.Collection [].map.call indexes, (char) ->
+                char: char
+            @hasContacts = false
 
         return unless @collection
 
         pattern = require('config').search.pattern 'tag'
         @tag = app.model.get('filter')?.match(pattern)?[1]
 
-        @hasContacts = @collection.model is ContactViewModel
-
-        if @hasContacts
-            @listenTo @,
-                'render': -> setTimeout => @children.call 'lazyLoadAvatar'
+        # if @hasContacts
+        #     @listenTo @,
+        #         'render': -> setTimeout => @children.call 'lazyLoadAvatar'
+        #
+        #     @listenTo @collection,
+        #         'update': @toggleEmpty
+        #
+        # @listenTo app.vent,
+        #     'filter:tag': (value) ->
+        #         @tag = value
+        #         @render() if @hasContacts
+        #         @triggerMethod 'filter'
+        #     'filter:text': (value) ->
+        #         @render() if app.model.get 'scored'
+        #         @triggerMethod 'filter'
+        #
 
-            @listenTo @collection,
-                'update': @toggleEmpty
 
-        @listenTo app.vent,
-            'filter:tag': (value) ->
-                @tag = value
-                @render() if @hasContacts
-                @triggerMethod 'filter'
-            'filter:text': (value) ->
-                @render() if app.model.get 'scored'
-                @triggerMethod 'filter'
+    serializeData: ->
+        _.extend {}, super, indexed: @indexed
 
 
     toggleEmpty: ->
-        @$el.toggleClass 'empty', not @children.length
+        @el.classList.toggle 'empty', not @children.length
+
+
+    updateTagClasses: ->
+        @el.removeAttribute 'class'
+        @toggleEmpty()
+
+        models = @collection.indexes.get(@model.get 'char')
+        tags = _.chain models.map (vmodel) -> vmodel.get 'tags'
+            .flatten()
```

Comment:
Yup, unfortunately, flapMap is available since 4.0.0, and we're stuck to v3.10 due to Marionette incompatibilities :expressionless: (which'll be solved in Marionette 3.0)